### PR TITLE
ZK-4688: A null parameter passed to a global command causes ClassCastException

### PR DESCRIPTION
**Pull request by DevChu**

### CODE REVIEW

File: zkbind/src/org/zkoss/bind/impl/ParamCall.java [hunk 1]
```diff
@@ -208,8 +209,16 @@ private Object resolveParameter(Annotation[] parmAnnos, Class<?> paramType, Stri
...
-		if (val == null)
+		if (val == null && _bindingArgs != null && index < _bindingArgs.size()) {
+			int argIndex = 0;
+			for (Map.Entry<String, Object> entry : _bindingArgs.entrySet()) {
+				// skip using positional if the param key is not auto-generated
+				if (argIndex == index && !entry.getKey().startsWith(Parser.SIMPLIFIED_COMMAND_PARAM_PREFIX))
+					return null;
+				argIndex++;
+			}
 			val = resolvePositionalParameter(paramType, index);
```
Comment (scribetw) [inline zkbind/src/org/zkoss/bind/impl/ParamCall.java:220]: move to resolvePositionalParameter

File: zktest/src/org/zkoss/zktest/test2/B95_ZK_4688VM.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+public class B95_ZK_4688VM {
+
+	private String result = "";
+
+	@GlobalCommand
+	@NotifyChange("result")
+	public void stopSetup(@BindingParam("finishDate") Date when, @BindingParam("setupQty") int setupQty,
+			@BindingParam("reason") Locale reason) {
+		result = when + "," + setupQty + reason;
```
Comment (scribetw) [inline zktest/src/org/zkoss/zktest/test2/B95_ZK_4688VM.java:23]: add ","

**Review by scribetw — CHANGES_REQUESTED** [review]

File: zkbind/src/org/zkoss/bind/impl/ParamCall.java [hunk 3]
```diff
@@ -218,14 +219,14 @@ private Object resolveParameter(Annotation[] parmAnnos, Class<?> paramType, Stri
...
-					val = v;
+			int argIndex = 0;
+			for (Map.Entry<String, Object> entry : _bindingArgs.entrySet()) {
+				// skip using positional if the param key is not auto-generated
+				if (argIndex == index) {
+					if (entry.getKey().startsWith(Parser.SIMPLIFIED_COMMAND_PARAM_PREFIX))
+						val = entry.getValue();
 					break;
 				}
-				i++;
```
Comment (scribetw) [inline zkbind/src/org/zkoss/bind/impl/ParamCall.java:228]: ++

**Review by plijyfes — APPROVED** [review]

**Review by scribetw — CHANGES_REQUESTED** [review]

**Review by scribetw — APPROVED** [review]

**Comment by scribetw** [comment]
@zkadmin rebase

**Comment by zkadmin** [comment]
I just pushed up the changes, enjoy!

**Pull request merged**